# IC and RankIC Analysis

This notebook evaluates the predictive power of the momentum factor using:

- Information Coefficient (IC)
- Rank Information Coefficient (RankIC)

Data source: Yahoo Finance (AAPL)

In [1]:
import pandas as pd
import numpy as np
import yfinance as yf
from scipy.stats import spearmanr

In [2]:
# Download data

data = yf.download("AAPL", start="2023-01-01", end="2024-01-01")

# Flatten columns if needed
if isinstance(data.columns, pd.MultiIndex):
    data.columns = data.columns.get_level_values(0)

# Momentum factor
data["Momentum_20"] = data["Close"].pct_change(20)

# Future 5-day return
data["Future_Return_5D"] = data["Close"].shift(-5) / data["Close"] - 1

# Remove missing values
ic_data = data[["Momentum_20", "Future_Return_5D"]].dropna()

# IC
ic = ic_data["Momentum_20"].corr(ic_data["Future_Return_5D"])

print("Information Coefficient (IC):", round(ic, 4))

[*********************100%***********************]  1 of 1 completed

Information Coefficient (IC): -0.1354


In [3]:
rank_ic, p_value = spearmanr(
    ic_data["Momentum_20"],
    ic_data["Future_Return_5D"]
)

print("RankIC:", round(rank_ic, 4))

RankIC: -0.1226


## Results Interpretation

The momentum factor produced an IC of -0.1354 and a RankIC of -0.1226.

Both values are negative, indicating that higher momentum scores were associated with lower future 5-day returns during the sample period.

This suggests that the 20-day momentum factor had weak and negative predictive power for AAPL in 2023 and may not be an effective standalone factor for this stock.